### ISSUE TITLE
S3 Classes with format and print?

### ISSUE THREAD

**Issue by krlmlr** [gaborcsardi/prettyunits#5]
For instance:

```
dput(pretty_bytes(1000))
> structure(1000, class = "pretty_bytes")
print(pretty_bytes(1000))
> [1] "1 kB"
```

Would you support this kind of functionality? It might break compatibility, but then there are not many revdeps (yet) ;-)

**Comment by gaborcsardi** [comment]
Yeah, that would be a more R-ish way, I guess. Which also means somewhat more messy of course. :)
E.g. if you do 

```r
x <- structure(1337, class = "pretty_bytes")
paste("File size:", x)
```
then no formatting is used. If we also implement `as.character.pretty_bytes`, then even this works, but it probably just messes up something else.

But we can do it, anyway, and then we'll see how it works out. :)

**Comment by krlmlr** [comment]
How important is backward compatibility? We could start by providing

```
pretty <- function(x, units) {
  structure(x, class = "pretty", units = units)
}
```

and deprecating the `pretty_...` stuff, just to reintroduce it in version 2.0.0. Or we break everything right away ;-)

**Comment by gaborcsardi** [comment]
I would like to keep the current way as well. We can just add a `pretty` or maybe `pretty_units` class independently I think. The current way would be kept as it is, so nothing breaks. Yeah, no many revdeps on CRAN, but I do use it in a couple of packages, and I also like the current simple way.

**Comment by gaborcsardi** [comment]
We probably want to use vctrs for this now?

**Comment by krlmlr** [comment]
I think subclassing `"vctrs_vctr"` will help us a lot here.
